base/src/main/java/com/smartdevicelink/managers/file/BaseFileManager.java

```diff
@@ -329,6 +329,16 @@ public void uploadFile(@NonNull final SdlFile file, final CompletionListener lis
 			listener.onComplete(true);
 			return;
 		}
+		// In sdl_ios: HAX: [#827](https://github.com/smartdevicelink/sdl_ios/issues/827) Older versions of Core had a bug where list files would cache incorrectly.
+		// This led to attempted uploads failing due to the system thinking they were already there when they were not.
+		if (!file.isPersistent() && !hasUploadedFile(file)) {
+			file.setOverwrite(true);
+		}
+		if (!file.isOverwrite() && hasUploadedFile(file)) {
+			Log.w(TAG, "Files that have already uploaded and have an Overwrite property set to false will not be re-Upload");
```

Comment:
```suggestion
			og.w(TAG, String.format("%s has already been uploaded and the overwrite property is set to false. It will not be uploaded again", file.getName()));
```

Reply:
Can we also update the static icon warning message to show the file name?